src/adapter/etl-adapter-parquet/src/Flow/ETL/Adapter/Parquet/ParquetLoader.php

```diff
@@ -52,9 +56,14 @@ public function __unserialize(array $data) : void
 
     public function closure(Rows $rows, FlowContext $context) : void
     {
-        $this->writer($context)->close();
+        if (\count($this->writers)) {
```

Comment:
No need to count, foreach handles that already.

Reply:
https://github.com/flow-php/flow/issues/713